src/runtime/virtcontainers/qemu.go

```diff
@@ -710,6 +710,30 @@ func (q *qemu) setupVirtiofsd(ctx context.Context) (err error) {
 	return err
 }
 
+func (q *qemu) stopVirtiofsd(ctx context.Context) (err error) {
+
+	// kill virtiofsd
+	if q.state.VirtiofsdPid == 0 {
+		return errors.New("invalid virtiofsd PID(0)")
```

Comment:
Doesn't it mean that it's been either killed or no started yet?  I'm wondering whether silently returning `nil`, after ensuring the socket doesn't exist / got removed, would be reasonable in this case.

Reply:
Yeah. I was modeling after how it is handled in CLH case today. Realistically, the socket will get removed anyway as part of the filesystem cleanup (ie /var/run/vc/vm/sandbox-id is removed entirely anyway), but its probably a good idea to try to remove in either case.

... Mulling ...

Reply:
@fidencio WDYT about keeping as is, and fixing as part of refactoring to leverage virtiofsd interface?

I'd like it consistent in implementation between CLH and QEMU. As a next step I'm recommending we update QEMU to use virtiofsd interface, and we can clean up / refactor for both VMMs at this point?

Reply:
sounds good to me, @egernst.

Reply:
Issue to track: https://github.com/kata-containers/kata-containers/issues/1757